Minor fixes and improvements to media players and lights.

georgesak opened the pull request.

georgesak committed:
- Minor fixes.

georgesak committed:
- Add async_set_(digital/analog/serial) which allows automation to change the value of joins on the fly.
  Example from a button tap taht sets the analog join 352 to value of 2:
  tap_action:
      action: call-service
      service: crestron.set_analog
      data:
        join: 51
        value_join: 2

georgesak committed:
- Changes to the media player:
  - Add volume up/down (pulses a digital join)
  - Changes the mute to be a toggle (now pulses the digital join). This is needed for my own system, change back otherwise.
  - Add a power on command that defaults to input 1

georgesak committed:
- Improvements to lights:
  - Change the divider to 257, allowing full range (0-255 -> -=65535)
  - Add an default brightness per light (otherwise default to 50%)

georgesak committed:
Update README.md

georgesak committed:
Update README.md

georgesak committed:
Update README.md

georgesak committed:
Update README.md

georgesak committed:
Update README.md

georgesak committed:
Update README.md

georgesak committed:
- Add unique IDs to all device types, which allows modifications from the UI.

georgesak committed:
Merge branch 'master' of https://github.com/georgesak/home-assistant-crestron-component

georgesak committed:
- Fix unique ID generation:
  - Use slugify to generate cleaner names
  - Use the DOMAIN instead of HUB (which is not stable)

sebastiensoulac commented:
Hello @georgesak 

Thanks for the update on crestron integration with HA.
I would like to control my C2N-Amp crestron with your integration.

Can you help me to do this integration?
If yes, can you tell me how to fill the join on the crestron.
my yaml config:
`crestron:
  port: 16384
media_player:
  - platform: crestron
    name: "Room 1"
    turn_off: 10
    turn_on: 9
    mute_join: 4
    volume_up_join: 1
    volume_down_join: 2
    volume_join: 1
    source_number_join: 13
    sources:
      1: "Android TV"
      2: "Roku"
      3: "Apple TV"
      4: "Chromecast"
      7: "Volumio"
      8: "Crestron Streamer"`

The source names are only indicative.

I would like to have zone On/Off and volume management.
Thanks for your help

georgesak commented:
> Hello @georgesak
> 
> Thanks for the update on crestron integration with HA. I would like to control my C2N-Amp crestron with your integration.
> 
> Can you help me to do this integration? If yes, can you tell me how to fill the join on the crestron. my yaml config: `crestron: port: 16384 media_player:
> 
> * platform: crestron
>   name: "Room 1"
>   turn_off: 10
>   turn_on: 9
>   mute_join: 4
>   volume_up_join: 1
>   volume_down_join: 2
>   volume_join: 1
>   source_number_join: 13
>   sources:
>   1: "Android TV"
>   2: "Roku"
>   3: "Apple TV"
>   4: "Chromecast"
>   7: "Volumio"
>   8: "Crestron Streamer"`
> 
> The source names are only indicative.
> 
> I would like to have zone On/Off and volume management. Thanks for your help

Hi, please open an issue instead, will try to help there. I don't have experience with the C2N-Amp but as long as you have a processor (2-series or above), it should work.

sebastiensoulac commented:
Thank you 
I have created a problem as requested

Ryamonster10 committed:
Fixed Unique IDs on Shade Componets

georgesak committed:
Update README.md

Add note about broken entities.

georgesak committed:
Update README.md

georgesak committed:
Merge pull request #1 from Ryamonster10/patch-1

Fixed Unique IDs on Shade Componets

georgesak committed:
Updates to the light component:
- Remove light type, it was not useful. We always default to a dimmable light.
- Remove deprecated supported_features and replace with attr_color_mode and attr_supported_color_modes
- Other minor fixes

georgesak committed:
- Updated cover.py to remove deprecated code, should now work with newest HA versions (thank you ddetton@ for your help in identifying and testing the issue!).

justinvallely committed:
Uses the new ClimateEntityFeature enum instead of the deprecated support constants

georgesak committed:
Merge pull request #9 from justinvallely/master

Uses the new ClimateEntityFeature enum

georgesak committed:
- Remove deprecated SUPPORT_* constants and replace with MediaPlayerEntityFeature (this broke with 2025.10 release).
